C++/merge sort.cpp

```diff
@@ -0,0 +1,57 @@
+#include<iostream>

```

Comment:
We already have merge sort implementation in cpp. Please check readme for what algorithms you can implement.